src/main/java/gr/gousiosg/javacg/stat/JCallGraph.java

```diff
@@ -149,6 +213,113 @@ public static void main(String[] args) {
+
+    listOfFilteredClasses = getFilteredClassesFromJar(listOfFilteredClasses, className);
+
+    if(listOfFilteredClasses.size() > 1) {
+      LOGGER.error("Multiple class instances found as listed below:- ");
+      for(JarEntry entry : listOfFilteredClasses)
+        LOGGER.error(entry.getName());
+      System.exit(1);
+    }
+    if(listOfFilteredClasses.size()==0){
+      LOGGER.error("no class instances found ");
+      System.exit(1);
+    }
+
+    return fetchMethodSignatures(jarFile, listOfFilteredClasses.get(0), methodName, returnType, parameterTypes);
+
+  }
+
+  //Fetch JarEntry of all classes in a Jan using JarInputStream
+  public static ArrayList<JarEntry> getAllClassesFromJar(JarInputStream JarInputStream) throws IOException {
+    JarEntry Jar;
+    ArrayList<JarEntry> listOfAllClasses = new ArrayList<>();
+    while(true) {
+      Jar = JarInputStream.getNextJarEntry();
+      if(Jar == null)
+        break;
+      if((Jar.getName().endsWith(".class")))
+        listOfAllClasses.add(Jar);
+    }
+    return listOfAllClasses;
+  }
+
+  //Fetch filtered classes from a list of JarEntry
+  public static ArrayList<JarEntry> getFilteredClassesFromJar(ArrayList<JarEntry> listOfAllClasses, String className) {
+    listOfAllClasses = listOfAllClasses.stream().filter(e -> e.getName().endsWith(className)).collect(Collectors.toCollection(ArrayList::new));
+    return listOfAllClasses;
+  }
+public static ArrayList<Pair<String, String>> fetchAllMethodSignaturesForyaml (JarFile JarFile,JarEntry Jar) throws IOException {
+  ClassParser cp = new ClassParser(JarFile.getInputStream(Jar), Jar.getName());
+  JavaClass jc = cp.parse();
+
+  Method[] methods = jc.getMethods();
+  ArrayList<Pair<String, String>> signatureResults = new ArrayList<>();
+  for(Method tempMethod : methods)
+    if(Arrays.stream(tempMethod.getAnnotationEntries()).filter(e->e.getAnnotationType().equals("Lorg/junit/Test;")).toArray().length > 0)
+      signatureResults.add(new Pair<>(jc.getClassName().substring(jc.getClassName().lastIndexOf(".")+1)+"#"+tempMethod.getName(),jc.getClassName() + "." + tempMethod.getName() + tempMethod.getSignature()));
+
+
+  return signatureResults;
+}
+  //Fetch the method signature of a method from a JarEntry
+  public static String fetchMethodSignatures(JarFile JarFile, JarEntry Jar, String methodName, Optional<String> returnType, Optional<String> paramterTypes) throws IOException {
+    ClassParser cp = new ClassParser(JarFile.getInputStream(Jar), Jar.getName());
+    JavaClass jc = cp.parse();
+
+    Method[] methods = jc.getMethods();
+    ArrayList<Method> signatureResults = new ArrayList<>();
+
+    for(Method tempMethod : methods)
+      if(tempMethod.getName().equals(methodName))
+        signatureResults.add(tempMethod);
+
+    if(returnType.isPresent()) {
+      signatureResults = signatureResults.stream().filter(e -> e.getReturnType().toString().contains(returnType.get())).collect(Collectors.toCollection(ArrayList::new));
```

Comment:
Refactor:
```
.stream()
   .map(e -> e.getReturnType())
   .map(e -> e.toString())
   .filter(e -> e.contains(returnType.get())
   .findFirst();
```

This gives you an `Optional<Method>`, but you only use `signatureResults.get(0)` below, so it should be equivalent.